packtools/sps/validation/related_articles.py

```diff
@@ -57,31 +57,28 @@ def related_articles_matches_article_type_validation(self, correspondence_list=N
             raise ValidationRelatedArticleException("Function requires a list of dictionary with article type and related article types")
 
         for related_article in self.related_articles:
-            validated = self.article_type in correspondence_dictionary and correspondence_dictionary.get(self.article_type) == related_article.get('related-article-type')
-            result.append(
-                {
-                    'title': 'Related article type validation',
-                    'xpath': './article[@article-type] .//related-article[@related-article-type]',
-                    'validation_type': 'match',
-                    'response': 'OK' if validated else 'ERROR',
-                    'expected_value': 'A valid match such as the following: {}'.format(
-                        ' or '.join([f'({article}, {related})' for article, related in correspondence_dictionary.items()])
-                    ),
-                    'got_value': 'article-type: {}, related-article-type: {}'.format(
-                        self.article_type,
-                        related_article.get('related-article-type')
-                    ),
-                    'message': 'Got {}, expected {}'.format(
-                        f'({self.article_type}, {related_article.get("related-article-type")})',
-                        f' or '.join([f'({article}, {related})' for article, related in correspondence_dictionary.items()])
-                    ),
-                    'advice': None if validated else 'Consider replacing the article-type and related-article-type '
-                                                     'with a tuple from the following list: {}'.format(
-                        f' or '.join([f'({article}, {related})' for article, related in correspondence_dictionary.items()])
-                    )
-                }
-            )
-        return result
+            for item in correspondence_list:
+                if isinstance(item, dict) and item.get('article-type') == self.article_type:
```

Comment:
estas duas linhas devem ficar fora do for da linha 59, pois é suficiente você executar uma única vez, porque o xml só terá 1 `article/@article-type` que não precisa ser feito para cada related-article do XML

Reply:
OK @robertatakenaka